### PR TITLE
Remove flask and brainzutils dependency from listenbrainz_spark

### PR DESCRIPTION
**Pull request by amCap1712**
Flask and in turn brainzutils was being used in listenbrainz_spark only for the purpose of loading configuration values and logging. We instead use the logging module. Further, we add PysparkIntegration and PysparkWorkerIntegraton to the sentry setup in listenbrainz_spark. This may come handy in the production setup.

### TODO before merging
1) #1271 should be merged prior to this
2) The PysparkWorkerIntegration needs to be tested to make sure that at the least it does not break production setup.

### CODE REVIEW

**Review by mayhem — COMMENTED** [review]
Ok, looks pretty good to me. There is one spark test that needs fixing. Once you're done with that, coordinate with me about doing a test deployment to ensure that everything is cool before we merge this.

File: listenbrainz_spark/config.py.sample [hunk 1]
```diff
@@ -19,10 +19,10 @@ SPARK_RESULT_QUEUE = "spark_result"
 # calculate stats on X months data
 STATS_CALCULATION_WINDOW = 1
 
-LOG_SENTRY = {
-    'dsn':'',
-    'environment': 'development',
-}
+#LOG_SENTRY = {
+#    'dsn':'',
+#    'environment': 'development',
```
Comment (alastair) [inline listenbrainz_spark/config.py.sample:24]: how do we set config.py in production? In that case, we need to make sure that we set environment properly there

File: listenbrainz_spark/ftp/__init__.py [hunk 2]
```diff
@@ -20,7 +19,7 @@ def connect(self):
             self.connection = ftplib.FTP(config.FTP_SERVER_URI)
             self.connection.login()
         except ftplib.error_perm:
-            current_app.logger.critical("Couldn't connect to FTP Server, try again...")
+            logging.critical("Couldn't connect to FTP Server, try again...")
```
Comment (alastair) [inline listenbrainz_spark/ftp/__init__.py:22]: if we are going to use python logging outside of flask then we should make some effort to configure it.
It'd be good to be able to set a level in config.py. Additionally, we should have a logging setup method that allows us to configure the handlers that we want (and the ability to add a file handler if needed). We could put this in a separate py file (or listenbrainz_spark/init) and use it in all places that we do logging

**Review by alastair — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -31,7 +31,7 @@ WORKDIR /code
 
 RUN mkdir /code/listenbrainz
 WORKDIR /code/listenbrainz
-RUN pip3 install pip==20.2.4
+RUN pip3 install pip==21.0.1
```
[the remark this answers is not in the document]
Comment (alastair) [reply Dockerfile:34]: oh, I see that it includes the commits for that PR too, no problem!

File: listenbrainz_spark/ftp/__init__.py (hunk 2 above)
Comment (amCap1712) [reply listenbrainz_spark/ftp/__init__.py:22]: That's a great idea! I'll get to it.

File: listenbrainz_spark/config.py.sample [hunk 4]
```diff
@@ -19,10 +19,10 @@ SPARK_RESULT_QUEUE = "spark_result"
...
+#}
 SENTRY_DSN_PUBLIC = ''
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply listenbrainz_spark/config.py.sample:26]: SENTRY_DSN_PUBLIC is being set in production. I checked found that the LOG_SENTRY.dsn value contains the secret part and the SENTRY_DSN_PUBLIC does not. However, the secret part is optional and deprecated in newer versions of sentry. This is explained in the [sentry docs](https://docs.sentry.io/product/sentry-basics/dsn-explainer/#what-the-dsn-does). We can remove the SENTRY_PUBLIC_DSN variable.

During testing this and other projects with `bu-sentrysdk-upgrade`,  we used the DSN without the secret part. So, it should be safe to remove secret part from the production configurations as well.

File: listenbrainz_spark/config.py.sample (hunk 1 above)
Comment (amCap1712) [reply listenbrainz_spark/config.py.sample:24]: We do set the environment in production to `production`.

**Comment by alastair** [comment]
@amCap1712 can you rebase this?

**Pull request closed without merging**